### ISSUE TITLE
Reset current user pointer when the object is destroyed

### ISSUE THREAD

**Issue by mescalito** [parse-community/Parse-SDK-JS#169]
Why a {code: 209, error: "invalid session token"} after a user.destroy and then singUp() a new user?

``` javascript
$scope.doesNotExist = function (value) {
            var db = ['john.doe@mail.com', 'jane.doe@mail.com'];

            var deferral = $q.defer();
            var user = new Parse.User();
            user.set("username", "clave9");
            user.set("password", "my pass");
            user.signUp(null, {
                success: function(user) {
                    // Hooray! Let them use the app now.
                    console.log("success!");
                    //resolve();
                    // holly shit! now I have to destroy the error and wait for the whole form to be submmited.
                    user.destroy({
                        success: function(myObject) {
                            // The object was deleted from the Parse Cloud.
                            console.log("destroy!!!!!!!!!!!");
                            deferral.resolve();
                        },
                        error: function(myObject, error) {
                            // The delete failed.
                            // error is a Parse.Error with an error code and message.
                            console.log("failed destroy!!!!!!!!!!!");
                            deferral.resolve();
                        }
                    });
                    Parse.User.logOut();
                    deferral.resolve();
                },
                error: function(user, error) {
                    // Show the error message somewhere and let the user try again.
                    console.log("Errrrrror: " + error.code + " " + error.message);
                    //reject();
                    deferral.reject("AUTH_REQUIRED");
                }
            });
            return deferral.promise;
        };
```

This issue is related to this question in StackOverflow: http://stackoverflow.com/questions/34823496/how-do-i-check-if-a-username-is-taken-with-parse-com-when-all-user-aclpublic-re

**Comment by andrewimm** [comment]
When you call signup, you set the current user to be that User object. Upon destroying the user, you still have a (now-invalid) user object stored as the current user. You'll need to log out after destroying the user; otherwise, your JS SDK will continue passing on an invalid session token with every request.

Obviously, this isn't ideal behavior, so we should consider overriding User.destroy to check if the object being destroyed is the current user, and automatically clearing the current user pointer in that case.

**Comment by hanoii** [comment]
I just got here after a test and I think what @andrewimm makes sense. Right now though, something's odd is happening, if i destroy a user and then call logout, logout fails with 400 bad request with an invalid token error. So what would be the correct way of doing a destroy, then logout right now?

@mescalito Mescalito is my goto nickname mostly everywhere.. funny I found you on github on this issue :)
